tests/topotests/lib/topogen.py

```diff
@@ -771,6 +771,10 @@ def __init__(self, tgen, cls, name, **params):
         # Mount gear log directory on a common path
         self.net.bind_mount(self.gearlogdir, "/tmp/gearlogdir")
 
+        # Ensure pid file
+        with open(os.path.join(self.logdir, self.name + ".pid"), "w") as f:
+            f.write(str(tgen.net.hosts[self.name].pid))
```

Comment:
Should add a newline to the end of the file. Not having it can very occasionally cause issues in scripts/editors. Also `self.net` --> `tgen.net.hosts[self.name]` here so:
```
    f.write(str(self.net.pid) + "\n")
```

Reply:
Updated. Thanks.